docs/ca/dev/monitoring/opentelemetry-instrumentation.md

```diff
@@ -0,0 +1,442 @@
+This will add a error event into the root span and will change its status to the `error` one. We would highly recommend to check this part during integration of OpenTelemetry into your system.
+
+## Changing a service name
+
+Service name allows you to filter traces by the source. You may want to check only Yves traces or mb just to check only Glue requests, but you backend can show you a bunch of CLI commands from your Scheduler.
+You can define a service name via `\Spryker\Service\Monitoring\MonitoringService::setApplicationName()` or `\Spryker\Service\Opentelemetry\OpentelemetryService::setResourceName()`.
+If you are using the `MonitoringService` and any method from it was executed, you will get a generated service name in any case. By default it looks like `APPLICATION-REGION_OR_STORE(application.env)`, where `APPLICATION` is a name of your application (ZED, YVES, GLUE, etc.), `REGION_OR_STORE` is a current store or region name that depends if your application works in Dynamic Store mode or not, and `application.env` is an env name from your deploy file. You can change any of those value via `MonitoringService` or change the name completely via `OpentelemetryService`.
+If nothing was provided by services, OpenTelemetry integration will try to resolve a service name for you.
+First of all `OTEL_SERVICE_NAME_MAPPING` value will be fetched and will try to find a proper service name based on the URL or CLI binary file name.
+If no value was provided or nothing from it was suitable or us - the default service name from `OTEL_DEFAULT_SERVICE_NAME` env value will be used.
+
+## Changing a trace name
+
+Trace name (or in our case a root span name) should show you what request or command was executed. By default it should include a HTTP method name with an URL for the WEB requests and command name for a command execution.
+But it can be also changed during the execution. OOTB Spryker will change a WEB requests trace name in order to reflect a route name of the request. But if you don't like it or don't use a `Monitoring` module, you can define it via `\Spryker\Service\Opentelemetry\OpentelemetryService::setRootSpanName()` or `\Spryker\Service\Monitoring\MonitoringService::setTransactionName()` accordingly.
+
+## Recommendations
+
+Due to the fact that we use PHP code to instrument codebase, you should consider performance. Tracing is an expensive operation and we should use it wisely.
+
+Please minimise amount of generated spans per request. OpenTelemetry documentation recommends to have no more than 1000 of them. So you can skip some spans via configuration that are not relevant to you. Don't be afraid, errors will be processed even if the method was not instrumented because Error Event will be attached to the root span.
+
+Use sampling to not get a full trace every time. Please check configuration section for the reference.
```

Comment:
Provide a link to the section where it's explained

Reply:
https://github.com/spryker/spryker-docs/pull/3050/files/b1d946b28510f9cf0536ad1074da8b78b2237bf4#diff-36d4903a40a4efb31f615e2c34b30091fe1534f17550dacc50415173a7ff052eR371